contracts/hooks/examples/FullRange.sol

```diff
@@ -336,17 +330,17 @@ contract FullRange is BaseHook, ILockCallback {
 
         uint160 newSqrtPriceX96 = (
             FixedPointMathLib.sqrt(
-                FullMath.mulDiv(uint128(-balanceDelta.amount1()), FixedPoint96.Q96, uint128(-balanceDelta.amount0()))
+                FullMath.mulDiv(uint128(balanceDelta.amount1()), FixedPoint96.Q96, uint128(balanceDelta.amount0()))
             ) * FixedPointMathLib.sqrt(FixedPoint96.Q96)
         ).toUint160();
 
-        (uint160 sqrtPriceX96,,) = poolManager.getSlot0(poolId);
+        (uint160 sqrtPriceX96,,,) = poolManager.getSlot0(poolId);
 
         poolManager.swap(
             key,
             IPoolManager.SwapParams({
                 zeroForOne: newSqrtPriceX96 < sqrtPriceX96,
-                amountSpecified: MAX_INT,
+                amountSpecified: -MAX_INT,
```

Comment:
`- type(intX).max != type(intX).min` - ints are not symmetrical as theres 1 extra negative number than positive. So I think instead you might need a new `MIN_INT` variable to hold `type(int256).min`

Reply:
the minimum integer is actually `-MAX_INT - 1`

Reply:
good catch!